LanguageFeatures/Inline-classes/static_analysis_member_invocation_A03_t01.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion If Dm is a getter declaration with return type R then the static
+/// type of the invocation is [T1/X1 .. Ts/Xs]R.
+///
+/// @description Checks static type of a getter of an `inline` class
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/static_type_helper.dart";
+
+inline class IC1 {
+  final num id;
+  IC1(this.id);
+
+  int get asInt => id as int;
+}
+
+inline class IC2<T extends num> {
+  final T id;
+  IC2(this.id);
+
+  double get asDouble => id as double;
+}
+
+main() {
+  IC1 ic1 = IC1(42);
+  ic1.asInt.expectStaticType<Exactly<int>>();
+  ic1.id.expectStaticType<Exactly<num>>();
+
+  IC2 ic2 = IC2<double>(3.14);
+  ic2.asDouble.expectStaticType<Exactly<double>>();
+  ic2.id.expectStaticType<Exactly<num>>();
+
+  IC2<double> ic3 = IC2<double>(3.14);
```

Comment:
We might actually omit `<double>` from the initializing expression: Type inference on an inline class is potentially running some pieces of new code in the tools. The rest of the test (lines 39-40) should be unaffected because the type of `ic3` will still be `IC2<double>`, we just want to see if anything crashes when it encounters `IC2(3.14)`.